saleor/checkout/utils.py

```diff
@@ -1031,12 +1033,11 @@ def clear_delivery_method(checkout_info: "CheckoutInfo"):
         shipping_channel_listings=checkout_info.shipping_channel_listings,
     )
     if updated_fields:
-        checkout.save(
-            update_fields=updated_fields
-            + [
-                "last_change",
-            ]
-        )
+        updated_fields.append("last_change")
+
+    if save:
+        checkout.save(update_fields=updated_fields)
+    return updated_fields
```

Comment:
Should we perform a save only when we have any field to update? 
```suggestion
        updated_fields.append("last_change")

        if save:
            checkout.save(update_fields=updated_fields)
    return updated_fields
```